src/s1reader/s1_burst_slc.py

```diff
@@ -579,6 +585,332 @@ def az_carrier_components(self, offset, position):
+        else:
+            rad_lat = lat
+            rad_lon = lon
+
+        v = ellipsoid.a / np.sqrt(1 - ellipsoid.e2 * np.sin(rad_lat) * np.sin(rad_lat))
+
+        x = (v + hgt) * np.cos(rad_lat) * np.cos(rad_lon)
+        y = (v + hgt) * np.cos(rad_lat) * np.sin(rad_lon)
+        z = (v * (1 - ellipsoid.e2) + hgt) * np.sin(rad_lat)
+
+        return (x, y, z)
+
+    def az_fm_rate_mismatch_mitigation(self, path_dem: str, path_scratch: str,
+            threshold_rdr2geo = 1e-8,
+            numiter_rdr2geo = 25,
+            custom_radargrid = None):
+        '''
+        Calculate azimuth FM rate mismatch mitigation
+        Based on ETAD-DLR-DD-0008, Algorithm Technical Baseline Document.
+        Available: https://sentinels.copernicus.eu/documents/247904/4629150/
+        Sentinel-1-ETAD-Algorithm-Technical-Baseline-Document.pdf
+
+        Parameters:
+        -----------
+        path_dem: str
+            Path to the DEM to calculate the actual azimuth FM rate
+        path_scratch: str
+            Path to the scratch directory to store intermediate data
+        threshold_rdr2geo: int
+            Threshold of the iteration for rdr2geo
+        numiter_rdr2geo: int
+            Maximum number of iteration for rdr2geo
+        custom_radargrid_correction: isce3.product.RadarGridParameters
+            ISCE3 radar grid to define the correction grid
+        
+        Return:
+        -------
+        _: isce3.core.LUT2d
+            azimuth FM rate mismatch rate in radar grid in seconds.
+
+        '''
+        
+        os.makedirs(path_scratch, exist_ok=True)
+        
+        if custom_radargrid is None:
+            radargrid_correction = self.as_isce3_radargrid()
+        else:
+            radargrid_correction = custom_radargrid
+
+        # Define the correction grid from the radargrid
+        # Also define the staggered grid in azimuth to calculate acceeration
+        width_grid = radargrid_correction.width
+        length_grid = radargrid_correction.length
+        intv_t = 1/radargrid_correction.prf
+        intv_tau = (radargrid_correction.range_pixel_spacing * 2.0
+                    / isce3.core.speed_of_light)
+        delta_sec_from_ref_epoch = ((radargrid_correction.ref_epoch
+                                - self.orbit.reference_epoch).total_seconds()
+                                + radargrid_correction.sensing_start)
+        vec_t = np.arange(length_grid) * intv_t + delta_sec_from_ref_epoch
+        vec_t_staggered = (np.arange(length_grid+1)
+                           * intv_t
+                           + delta_sec_from_ref_epoch
+                           - intv_t / 2)
+
+        vec_tau = np.arange(width_grid)*intv_tau
+
+        grid_tau, grid_t = np.meshgrid(vec_tau, vec_t)
+
+        vec_position_intp = np.zeros((length_grid,3))
+        vec_vel_intp = np.zeros((length_grid,3))
+        vec_vel_intp_staggered = [None] * (length_grid+1)
+        for i_azimiuth, t_azimuth in enumerate(vec_t):
```

Comment:
```suggestion
        for i_azimuth, t_azimuth in enumerate(vec_t):
```
I am assuming the variable mispelling was not intentional